.github/workflows/deploy-generic.yml

```
@@ -0,0 +1,88 @@
+name: Deploy
+
+on:
+  workflow_call:
+    inputs:
+      env:
+        description: "which environment to deploy to"
+        required: true
+        type: string
+      ecr_repository:
+        description: "ecr repo to hold container image"
+        required: true
+        type: string
+      ecr_region:
+        description: "ecr region to hold container image"
+        required: true
+        type: string
+    secrets:
+      kube_namespace:
+        description: "the kubernetes namespace to deploy to"
+        required: true
+      kube_cert:
+        description: "cert used to verify identity to cluster"
+        required: true
+      kube_cluster:
+        description: "address of the cluster to connect to"
+        required: true
+      kube_token:
+        description: "used to authenticate to the cluster"
+        required: true
+      ecr_role_to_assume:
+        description: "role to authenticate ecr image repository push"
+        required: true
+      secret_key:
+        description: "secret key"
+        required: true
+      catalogue_token:
+        description: "token to authenticate with the catalogue"
+        required: true
+
+jobs:
+  deploy:
+    name: Deploy Helm Chart into Cloud Platform
+    environment: ${{ inputs.env }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # This is required for requesting the JWT
+      contents: read # This is required for actions/checkout
+    steps:
+      - uses: actions/checkout@v4
+      - uses: aws-actions/configure-aws-credentials@v2
```

Can you add a `name`, `id`, (and ideally a `shell` key if using the `run` key) to each step, and separate them with newlines? Example provided
```suggestion
      - name: Checkout
        id: checkout
        uses: actions/checkout@v4
      
      - name: Configure aws creds
        id: config-aws-creds
        uses: aws-actions/configure-aws-credentials@v2
```